examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,222 @@
+
+# %%
+# `model` is a machine learning pipeline made of the preprocessing stage
+# followed by a ridge regressor that will perform an internal cross-validation
+# at `fit` to select the best parameter `alpha`. The candidates for the
+# parameter `alpha` are given by the variable `alphas`.
+#
+# Cross-validation framework
+# --------------------------
+#
+# Before putting such a predictive model into production, one needs to evaluate
+# the performance of the model to have an idea of what to expect in production.
+#
+# Cross-validation should be used to make this analysis. First, it allows us to
+# quantify the variance of the model performance. A large variance of the score
+# metric will indicate that we cannot trust the reported performance nor try to
+# interpret findings built on internal model's parameters. One possible cause
+# of large variations are small sample sizes.
+from sklearn.model_selection import cross_validate
+from sklearn.model_selection import RepeatedKFold
+
+cv = RepeatedKFold(n_splits=10, n_repeats=10, random_state=0)
+cv_results = cross_validate(
+    model, X_train, y_train, cv=cv, return_estimator=True, n_jobs=2
+)
+
+# %%
+# Here, we used a repeated K-fold cross-validation. At each round of
+# cross-validation, it should be noted that the parameter `alpha` of the ridge
+# regressor is also optimized via another internal cross-validation. This
+# process is called a nested cross-validation and should always be implemented
+# whenever model's parameters need to be optimized.
+#
+# Analyze model performance
+# .........................
+#
+# As previously mentioned, one should look at the variance of the model
+# performance within the cross-validation framework.
+import matplotlib.pyplot as plt
+
+cv_score = cv_results["test_score"]
+plt.hist(cv_score, bins=200, density=True)
+plt.xlim([0, 1])
+plt.ylabel("Density")
+plt.xlabel("R2 score")
+_ = plt.title(
+    "Distribution of the scores on the test sets\n during the cross-validation"
+)
+
+# %%
+# We start by plotting the empirical distribution of the test score computed
+# during cross-validation.
+#
+# We observe little variation in R2 score and can thus safely use the results
+# to interpret the model performance.
+#
+# Our baseline model has an R2 of around 0.6 which is better than a dummy
+# regressor. We can therefore use this predictive model as a baseline against
+# more advanced machine learning pipelines.
+#
+# To conclude, cross-validation allows us to answer to two questions: are the
+# results reliable and, if it is, how good is the predictive model.
+#
+# Model inspection
+# ................
+#
+# Once we are happy or at least aware of any model limitations, we can
+# investigate the internals of our model. When we performed the
+# cross-validation, we set the parameter `return_estimator` to `True`.
+#
+# It allows us to get the different predictive models trained and tested within
+# cross-validation.
+cv_estimators = cv_results["estimator"]
```

Comment:
To make it explicit that we are cross-validating the full pipeline that comprises both the scaler and the regressor.

```suggestion
cv_pipelines = cv_results["estimator"]
```